File: 0682-baseball-game/0682-baseball-game.cpp
class Solution {
public:
    int calPoints(vector<string>& operations) {
        vector<int>st;
        int n = operations.size();
        for(int i = 0; i < n; i++){
            string o = operations[i];

            if(o == "C"){
                st.pop_back();
            }else if(o == "D"){
                st.push_back(2*st.back());
            }else if(o == "+"){
                st.push_back(st[st.size() - 1] + st[st.size()-2]);
            }else{
                st.push_back(stoi(o));
            }
        }
        int sum = 0;
        for(int i = 0; i < st.size(); i++){
            sum += st[i];
        }
        return sum;
    }
};

File: 0682-baseball-game/0682-baseball-game.java
class Solution {
    public int calPoints(String[] operations) {
        Stack<Integer>st = new Stack<>();

        int n = operations.length;
        for(int i = 0; i < n; i++){
            if(operations[i].equals ("+")){
                int top = st.pop();
                int newscore = top + st.peek();
                st.push(top);
                st.push(newscore);
            }
            else if(operations[i].equals("D")){
                
                int x = st.peek() * 2;
                st.push(x);
            }
            else if(operations[i].equals ("C")){
                st.pop();
            }else{
                st.push(Integer.parseInt(operations[i]));
            }
        }
        int ans = 0;
        while(!st.isEmpty()){
            ans += st.peek();
            st.pop();
        }
        return ans;
    }
}